mm2src/coins/eth.rs

```diff
@@ -1085,12 +1165,19 @@ impl MarketCoinOps for EthCoin {
 
     fn my_address(&self) -> Result<String, String> { Ok(checksum_address(&format!("{:#02x}", self.my_address))) }
 
-    fn get_public_key(&self) -> Result<String, MmError<UnexpectedDerivationMethod>> { unimplemented!() }
+    fn get_public_key(&self) -> Result<String, MmError<UnexpectedDerivationMethod>> {
+        let mut raw_pubkey = [0; 65];
+        raw_pubkey[0] = 0x04;
+        raw_pubkey[1..].copy_from_slice(self.key_pair.public());
+        let secp_public = PublicKey::from_slice(&raw_pubkey).unwrap();
```

Comment:
`hex::encode(raw_pubkey)` returns uncompressed version, this one is compressed. In order to compress the pubkey we need `PublicKey` instance and use `serialize` function.

If uncompressed pubkey is also fine, then we can simplify this function as:
```rs
let uncompressed_without_prefix = hex::encode(self.key_pair.public());
Ok(format!("04{}", uncompressed_without_prefix))
```

Otherwise we kind a need to use `PublicKey` here.

Reply:
> If uncompressed pubkey is also fine, then we can simplify this function as:

Yes, it's fine. I'm not sure if ETH uses compressed public keys at all. According to `ethereum_types` crate https://github.com/paritytech/parity-common/blob/master/ethereum-types/src/lib.rs#L20, returning the pubkey in uncompressed form will be a native way to work for Ethereum.